be/src/vec/olap/vgeneric_iterators.cpp

```diff
@@ -382,21 +399,36 @@ Status VMergeIterator::next_batch(vectorized::Block* block) {
         _block_row_locations.resize(block_row_max);
     }
     size_t row_idx = 0;
+    VMergeIteratorContext* pre_ctx = nullptr;
     while (block->rows() < block_row_max) {
         if (_merge_heap.empty()) break;
 
         auto ctx = _merge_heap.top();
         _merge_heap.pop();
 
         if (!ctx->need_skip()) {
```

Comment:
Is it possible to have different `need_skip` in a `VMergeIterator `? If all of `ctx` in each `VMergeIterator` is `need_skip` or not `need_skip`, maybe we should move this condition out of this while-loop

Reply:
need_skip is set in Comparator and it's used for every row